Util/openl-opentelemetry-javaagent-extension/src/org/openl/opentelemetry/javaagent/extension/ExecutableRulesMethodSpanGenerator.java

```diff
@@ -0,0 +1,107 @@
+package org.openl.opentelemetry.javaagent.extension;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.annotation.support.async.AsyncOperationEndSupport;
+import io.opentelemetry.instrumentation.api.instrumenter.util.ClassAndMethod;
+import io.opentelemetry.javaagent.bootstrap.Java8BytecodeBridge;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.bytecode.assign.Assigner;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.openl.rules.method.ExecutableRulesMethod;
+
+import java.lang.reflect.Method;
+import java.util.Set;
+
+import static io.opentelemetry.javaagent.bootstrap.Java8BytecodeBridge.currentContext;
+import static io.opentelemetry.javaagent.extension.matcher.AgentElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.namedOneOf;
+import static org.openl.opentelemetry.javaagent.extension.MethodSingletons.instrumenter;
+
+public class ExecutableRulesMethodSpanGenerator implements TypeInstrumentation {
+    private final Set<String> classNames;
+    private final Set<String> methodNames;
+
+    public ExecutableRulesMethodSpanGenerator(Set<String> classNames) {
+        this.classNames = classNames;
+        this.methodNames = Set.of("innerInvoke");
+    }
+
+    @Override
+    public ElementMatcher<TypeDescription> typeMatcher() {
+        return hasSuperType(namedOneOf(classNames.toArray(new String[0])));
```

Comment:
Why `classNames` is Set but nor an array? Why it is need there to converting Set to array instead using array directly?